### ISSUE TITLE
Document how exceptions are handled

### ISSUE THREAD

**Issue by ShaddyDC** [mystor/rust-cpp#109]
I'm evaluating different options to use C++ code from rust, and an important point for me is that error handling should be seamless. It would be convenient if what happens when an exception crosses the boundary was documented somewhere.
The only relevant information I found was #21.

If anybody else wonders as well, I've now tried it myself, and it results in this crash:
```rust
let result = unsafe {
    cpp!([] -> i32 as "std::int32_t" {
            throw std::runtime_error("oops");
            return 0;
        })
    };

```
```
fatal runtime error: Rust cannot catch foreign exceptions
Aborted (core dumped)
```

In hindsight, this is the obvious behaviour, but it's still useful to have it explicitly mentioned.

**Comment by ogoffart** [comment]
Right, as of now, nothing had been done to handle exceptions as they cross boundaries.

Since Rust 1.71, we could use `extern "C-unwind"`
But that would not result in the same behaviour as the rust runtime wouldn't know how to catch it.

**Comment by ogoffart** [comment]
@ebegumisa your comment seems sensible. a PR would be welcome.
